app/javascript/components/dosage/Dosage.js

```diff
@@ -174,9 +169,9 @@ class Dosage extends React.Component {
                 <DateInput
                   id="expiration_date"
                   date={this.state.expiration_date}
-                  onChange={date => this.setState({ expiration_date: date })}
+                  onChange={date => this.handleDateChange('expiration_date', date)}
                   placement="bottom"
-                  customClass="form-control-lg"
+                  customClass={!this.state.dosageInvalid ? 'form-control-lg' : 'form-control-lg is-invalid'}
                 />
```

Comment:
Should we have an error message about why the form is being blocked from submitting? It might be confusing if someone doesn't realize they put in bad dates.